promote-release-to-box.py

```diff
@@ -0,0 +1,290 @@
+import os
+import re
+import sys
+import json
+import yaml
+import argparse
+import requests
+import logging
+import subprocess
+from tqdm import tqdm
+from boxsdk import OAuth2,Client
+from artifactory import ArtifactoryPath
+from requests.auth import HTTPBasicAuth
+from boxsdk.exception import BoxAPIException
+
+
+logging.basicConfig(
+     level=logging.WARNING, 
+     format= '[%(asctime)s] {%(pathname)s:%(lineno)d} %(levelname)s - %(message)s',
+     datefmt='%H:%M:%S'
+ )
+
+def get_artifact_info_json(build_name, build_number, rt_auth = (None, None), rt_base_url = None):
+    ''' 
+    Expected Jfrog CLI is availble in the system.
+    Executes these:
+    1. jf config add orl-artifactory --interactive=false \
+        --enc-password=false --basic-auth-only \
+        --artifactory-url https://repo.orl.eng.hitachivantara.com/artifactory \
+        --password --user buildguy
+
+    2. jf rt search --server-id orl-artifactory \
+        --props "build.name=pdi-xxx-9.5.1.0;build.number=86" "*-9.5.1.0-86.zip" \
+        > builds.json
+
+    Alternative way: Use ArtifactoryBuildManager https://github.com/devopshq/artifactory#builds
+    
+    ^^^^^^^^^^^^^
+    from artifactory import ArtifactoryBuildManager
+
+    arti_build = ArtifactoryBuildManager(
+        orl_url, auth=auth_orl)
+
+    # Get all builds,space turns into %
+    all_builds = arti_build.builds
+    print(all_builds)
+    ^^^^^^^^^^^^^
+
+    '''
+    
+
+    # Define the command and arguments
+    command = [
+        'jf', 'config', 'add', 'orl-artifactory',
+        '--interactive=false', '--enc-password=false', '--basic-auth-only',
+        '--artifactory-url', f'{rt_base_url}',
+        '--password', f'{rt_auth[0]}',
+        '--user', f'{rt_auth[1]}'
+    ]
+
+    # Execute the command
+    subprocess.run(command)
+    
+    # Define the command and arguments
+    command = ['jf', 'rt', 'search', '--server-id', 'orl-artifactory', '--props', 
+               f'build.name={build_name};build.number={build_number}',
+               f'*' ]
+               # f'{artifact_name}']
+    
+    output_file = 'artifacts.json'
+
+    # Execute the command and capture the output
+    result = subprocess.run(command, capture_output=True, text=True)
+
+    # Parse the command output as JSON
+    output_json = json.loads(result.stdout)
+
+    # Save the JSON object to a file
+    with open(output_file, 'w') as file:
+        json.dump(output_json, file, indent=4)
+    
+    return output_json, set([artifact['path'].split('/')[-1] for artifact in output_json])
+
+
+def download_artifacts_v3(artifacts_to_release, builds_output_json, auth=(None, None), rt_base_url = None):
+    release_artifact_downloaded = []
+    for build_artifact in tqdm(builds_output_json):
+        file_name = build_artifact['path']
+        if build_artifact['path'].split('/')[-1] in artifacts_to_release.keys():
```

Comment:
You're comparing the search results you're getting from artifactory with the manifest. This deserves some comments here explaining the business rules.